packages/search-adapter/src/empathy/mappers/request/params/empathy-request-related-tags-query.mapper.ts

```diff
@@ -8,9 +9,30 @@ import { RequestMapper } from '../../../empathy-adapter.types';
  * @public
  */
 @injectable()
-export class EmpathyRequestRelatedTagsQueryMapper implements RequestMapper<QueryableRequest, string> {
-
+export class EmpathyRequestRelatedTagsQueryMapper
+  implements RequestMapper<QueryableRequest, string>
+{
   map({ relatedTags = [] }: QueryableRequest, query: string): string {
-    return relatedTags.reduce((chain, rt) => `${chain} ${rt.tag}`, query).trim();
+    const [leftRts, rightRts] = this.splitRelatedTagsByQueryPosition(
+      relatedTags,
+      query
+    );
+    return `${leftRts} ${query} ${rightRts}`.trim();
+  }
+
+  splitRelatedTagsByQueryPosition(
+    relatedTags: RelatedTag[],
+    query: string
+  ): string[] {
+    return relatedTags
+      .reduce(
+        ([left, right], rt) => {
+          return rt.query.indexOf(query) > rt.query.indexOf(rt.tag)
```

Comment:
Should we extract this to a method? `isQueryAfterTag` just to document what it does? I have to think for a few seconds, even minutes every time I read this code 😅

Reply:
I think we should normalise this query, or as a minimum, making it lowercase. In phones, the keyboard tends to auto-capitalises the typed text usually, and we would end up comparing non matching strings. You can try changing `lego` to `lEgo` in one of your tests to see how it fails.

Another option would be to just use each `relatedTag`, `tag` and `query` properties to determine if is a tag that should go to the left or to the right. This might be a simpler approach with a simpler condition, but I'm not sure if it is a 100% bulletproof.